.github/workflows/icfy-stats.yml

```diff
@@ -0,0 +1,53 @@
+name: Calculate ICFY stats
+
+on: [push]
+
+jobs:
+  build:
+    name: Build ICFY stats
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Set up Node
+      uses: actions/setup-node@v1
+      with:
+        node-version: '>=12.15.0'
+    - name: Checkout code
+      uses: actions/checkout@v2
+    - name: Fetch git history
+      run: git fetch --prune --unshallow
+    - name: Install dependencies
+      env:
+        PUPPETEER_SKIP_CHROMIUM_DOWNLOAD: true
+      run: npm ci
+    - name: Build ICFY stats
+      env:
+        NODE_ENV: production
+        BROWSERSLIST_ENV: defaults
+        WORKERS: 2
+      run: npm run analyze-icfy
+    - run: mkdir icfy-stats && mv client/{chart,stats}.json icfy-stats
+    - uses: actions/upload-artifact@v1
+      with:
+        name: icfy
+        path: icfy-stats
+    - name: Upload build artifact
+      env:
+        ICFY_SECRET: ${{ secrets.ICFY_SECRET }}
+      run: |
+        ANCESTOR_SHA1=$(git merge-base HEAD origin/master)
+        CURRENT_BRANCH=$(git rev-parse --abbrev-ref HEAD)
+        curl -X POST --globoff                                                     \
+          "http://api.iscalypsofastyet.com:5000/submit-stats?secret=$ICFY_SECRET"  \
+          -H 'Cache-Control: no-cache'                                             \
+          -H 'Content-Type: application/json'                                      \
+          -d '{
+                "payload": {
+                  "branch": "'"$CURRENT_BRANCH"'",
+                  "build_num": '"$GITHUB_RUN_NUMBER"',
```

Comment:
I'm a bit confused by the build numbering. For the feb1484 commit, the `build_num` received by ICFY was `16`:
```
Received CircleCI success webhook notification: { payload:
   { branch: 'update/move-icfy-stats-to-github-action',
     build_num: 16,
     sha: 'feb1484add484d02d57e6728bd5ad0adf7529698',
     ancestor: '0c9f34fbee2020e8d771dce3c98a1439c30324e6',
     from: 'gh-action' } }
```

But the run numbers in GH web UI are 9 figures starting with 4 and the artifact download URL is:
```
https://github.com/Automattic/wp-calypso/suites/480255681/artifacts/2208657
```

Reply:
Maybe this should be `GITHUB_RUN_ID`?

https://help.github.com/en/actions/configuring-and-managing-workflows/using-environment-variables

Reply:
Ugh. https://github.com/actions/upload-artifact/issues/50

Yeah, the run_id would let you query for artifacts: https://developer.github.com/v3/actions/artifacts/

Reply:
Now after the latest commit the build number is `17`, the build number is `44892545` and the artifact URL is `https://github.com/Automattic/wp-calypso/suites/480418004/artifacts/2211405`. Not quite there yet, but much closer 🙂

Reply:
```
curl https://api.github.com/repos/automattic/wp-calypso/actions/runs/44892545/artifacts
```
returns:
```
{
  "total_count": 1,
  "artifacts": [
    {
      "id": 2211405,
      "node_id": "MDg6QXJ0aWZhY3QyMjExNDA1",
      "name": "icfy",
      "size_in_bytes": 192025220,
      "url": "https://api.github.com/repos/Automattic/wp-calypso/actions/artifacts/2211405",
      "archive_download_url": "https://api.github.com/repos/Automattic/wp-calypso/actions/artifacts/2211405/zip",
      "expired": false,
      "created_at": "2020-02-25T13:04:29Z",
      "updated_at": "2020-02-25T13:04:30Z"
    }
  ]
}
```
I guess that with the build ID, getting to the artifact download URL became possible 🎉